### PR TITLE
Add MWL status to card zoom page

### PR DESCRIPTION
**Pull request by avandever**
This is intended to address #14 . What it does:

- Display if the card is restricted
- Display if the card is banned (with flexible wording in case it's needed)

What it doesn't do:

- Display rotation status

Questions I have:

1. formatting - I think it's terrible right now, but I'm not sure what will look better. Should I just play around with it a bit more, or are there guidelines I should apply?
2. Ban wording - would it make more sense to just say "banned" instead of worrying about the specific deck limit? That would be an easy change
3. i18n - Is there some framework for this I should be tapping into?

Screenshots:
screenshot 2017-10-13 15 07 09
screenshot 2017-10-13 15 07 18
screenshot 2017-10-13 15 07 24
screenshot 2017-10-13 15 07 30

### CODE REVIEW

**Comment by gereons** [comment]
I'd simply use the vocabulary from the MWL document: "restricted" and "removed". Also, it's probably useful to include the MWL version number.

**Comment by avandever** [comment]
New screenshots:
screenshot 2017-10-18 00 30 42
screenshot 2017-10-18 00 23 40
screenshot 2017-10-18 00 23 45

I still feel like there's got to be a better formatting option, maybe more similar to Rulings.

**Comment by avandever** [comment]
Yeah, I feel much better about this. Please let me know what you think.
screenshot 2017-10-18 00 44 49
screenshot 2017-10-18 00 44 55
screenshot 2017-10-18 00 45 02

**Comment by alsciende** [comment]
I like it a lot, thank you.

File: src/AppBundle/Service/CardsData.php [hunk 1]
```diff
@@ -738,6 +740,39 @@ function ($s) {
...
+        $response = array();
+        $card_code = (string) $card->getCode();
+        $now = new DateTime("now", new DateTimeZone('UTC'));
+        // L'ordre de tri est important pour identifier les règles actives
+        $mwls = $this->doctrine->getRepository('AppBundle:Mwl')->findBy(array(), array("dateStart" => "DESC"));
+        $found_current = false;
+        foreach ($mwls as $mwl) {
+            $mwl_cards = $mwl->getCards();
+            $card_mwl = $mwl_cards[$card_code] ?? null;
+            if ($card_mwl !== null) {
```
Comment (alsciende) [inline src/AppBundle/Service/CardsData.php]: On pourra aller un poil plus vite en faisant un `if (isset($mwl_cards[$card_code]))` ici et en créant la variable `$card_mwl` à l'instruction suivante.

File: src/AppBundle/Service/CardsData.php [hunk 2]
```diff
@@ -738,6 +740,39 @@ function ($s) {
         ));
     }
 
+    public function get_mwl_info($card) {
+        $response = array();
+        $card_code = (string) $card->getCode();
```
Comment (alsciende) [inline src/AppBundle/Service/CardsData.php]: `getCode` renvoie déjà un `string`, pas besoin de caster

File: src/AppBundle/Service/CardsData.php [hunk 3]
```diff
@@ -738,6 +740,39 @@ function ($s) {
...
+        // L'ordre de tri est important pour identifier les règles actives
+        $mwls = $this->doctrine->getRepository('AppBundle:Mwl')->findBy(array(), array("dateStart" => "DESC"));
+        $found_current = false;
+        foreach ($mwls as $mwl) {
+            $mwl_cards = $mwl->getCards();
+            $card_mwl = $mwl_cards[$card_code] ?? null;
+            if ($card_mwl !== null) {
+                $current = (!$found_current and $mwl->getDateStart() < $now);
+                if ($current) {
+                    $found_current = true;
```
Comment (alsciende) [inline src/AppBundle/Service/CardsData.php]: Pourquoi ne pas utiliser la propriété `Mwl.getActive()` ?

**Review by alsciende — CHANGES_REQUESTED** [review]
Sorry it took me several day to look at your PR. I like it, I'm just questioning why you didn't use the property Mwl.active.

File: src/AppBundle/Service/CardsData.php (hunk 3 above)
Comment (avandever) [reply src/AppBundle/Service/CardsData.php]: Doh! I hadn't run nrdb:legality:active-mwl, so it seemed it was either broken or intended for something else. :(

**Review by alsciende — APPROVED** [review]

**Comment by alsciende** [comment]
deployed!

**Comment by gereons** [comment]
Not sure if it's a server-side caching issue, but I only see info up to MWL 1.2.

**Pull request merged**